src/battle_script_commands.c

```diff
@@ -16383,3 +16317,78 @@ void BS_SetSnow(void)
     }
     gBattlescriptCurrInstr = cmd->nextInstr;
 }
+
+void BS_HandleMegaEvolution(void)
+{
+    NATIVE_ARGS(u8 battler, u8 caseId);
+
+    gActiveBattler = GetBattlerForBattleScript(cmd->battler);
+    HandleScriptMegaPrimalBurst(cmd->caseId, gActiveBattler, HANDLE_TYPE_MEGA_EVOLUTION);
+    gBattlescriptCurrInstr = cmd->nextInstr;
```

Comment:
might be useful to have some macros e.g. `#define NEXT_INSTR  gBattlescriptCurrInstr = cmd->nextInstr` and a similar FAIL_INSTR so we can put that at the end of these BS callnatives

Reply:
Since `gBattlescriptCurrInstr = cmd->nextInstr` is used by all 3 `various`, `callnative` and regular commands, it might be better to have this as a separate PR.